src/codec/decoder/audio.rs

```diff
@@ -142,25 +139,43 @@ impl AsMut<Context> for Audio {
 
 pub struct AudioFrameIter<'a, 'b> {
 	audio: &'a mut Audio,
-	packet: &'b mut packet::Packet,
+	_packet: &'b packet::Packet,
+	avpkt: AVPacket,
 	frame: frame::Audio,
 }
 
+impl<'a, 'b> AudioFrameIter<'a, 'b> {
+	fn new(audio: &'a mut Audio, packet: &'b packet::Packet) -> AudioFrameIter<'a, 'b> {
+		let avpkt = unsafe {
+			let mut avpkt: AVPacket = mem::zeroed();
+			ptr::copy(packet.as_ptr(), (&mut avpkt) as *mut AVPacket, 1);
+			avpkt
+		};
```

Comment:
Used unsafe code to make internal copy of AVPacket. This is to avoid `packet.clone()` which seems to be expensive.